src/backends/kimchi/mod.rs

```diff
@@ -704,7 +762,7 @@ impl Backend for KimchiVesta {
 
     fn assert_eq_var(&mut self, lhs: &KimchiCellVar, rhs: &KimchiCellVar, span: Span) {
         // TODO: use permutation to check that
-        self.add_generic_gate(
+        self.add_copy_constraint(
```

Comment:
you should be able to modify the wiring directly (in the same way `add_gate` adds information to the wiring, I'm not sure what is happening here :D

Reply:
Oh, I thought of doing it in a function so that we could call it in both the ```assert()``` calls, I'll do it directly instead then.

Reply:
I've added the wire modification directly in the functions. The tests are still failing though:
```
---- tests::examples::test_equals::case_1_kimchi_vesta stdout ----
thread 'tests::examples::test_equals::case_1_kimchi_vesta' panicked at src/backends/kimchi/prover.rs:99:26:
index out of bounds: the len is 15 but the index is 15

---- tests::examples::test_arithmetic::case_1_kimchi_vesta stdout ----
thread 'tests::examples::test_arithmetic::case_1_kimchi_vesta' panicked at src/backends/kimchi/prover.rs:99:26:
index out of bounds: the len is 3 but the index is 3

---- tests::examples::test_lc_return::case_1_kimchi_vesta stdout ----
thread 'tests::examples::test_lc_return::case_1_kimchi_vesta' panicked at src/backends/kimchi/prover.rs:99:26:
index out of bounds: the len is 3 but the index is 3

---- tests::examples::test_iterate::case_1_kimchi_vesta stdout ----
thread 'tests::examples::test_iterate::case_1_kimchi_vesta' panicked at src/backends/kimchi/prover.rs:99:26:
index out of bounds: the len is 4 but the index is 4

---- tests::examples::test_sudoku::case_1_kimchi_vesta stdout ----
thread 'tests::examples::test_sudoku::case_1_kimchi_vesta' panicked at src/backends/kimchi/prover.rs:99:26:
index out of bounds: the len is 22150 but the index is 22150
```

I'm getting out of bound errors. What could be the reason for this? I thought that maybe it's checking the number of copy constraints with gates and I'm not adding gates anymore so maybe that's why there's an error.